various-receipes/3rd-party-services-isolation/test/isolation.test.js

```diff
@@ -0,0 +1,118 @@
+const request = require("supertest");
+const sinon = require("sinon");
+const nock = require("nock");
+const {
+  initializeWebServer,
+  stopWebServer
+} = require("../../../example-application/api-under-test");
+const OrderRepository = require("../../../example-application/data-access/order-repository");
+
+let expressApp;
+let sinonSandbox;
+
+beforeAll(async (done) => {
+  // ️️️✅ Best Practice: Place the backend under test within the same process
+  expressApp = await initializeWebServer();
+  // ️️️✅ Best Practice: Ensure that this component is isolated by preventing unknown calls except for the Api-Under-Test
+  nock.disableNetConnect();
+  nock.enableNetConnect('127.0.0.1');
+
+  // ️️️✅ Best Practice: use a sandbox for test doubles for proper clean-up between tests
+  sinonSandbox = sinon.createSandbox();
+
+  done();
+});
+
+afterAll(async (done) => {
+  // ️️️✅ Best Practice: Clean-up resources after each run
+  await stopWebServer();
+  done();
+});
+
+beforeEach(() => {
+  // ️️️✅ Best Practice: Isolate the service under test by intercepting requests to 3rd party services
+  nock("http://localhost/user/").get(`/1`).reply(200, {
+    id: 1,
+    name: "John",
+  })
+
+  if (sinonSandbox) {
+    sinonSandbox.restore();
+  }
+});
+
+afterEach(() => {
+  // ️️️✅ Best Practice: Clean nock interceptors between tests
+  nock.cleanAll();
```

Comment:
Just curious - Why did you prefer afterEach over beforeEach?

Reply:
I don't know.. clean some something of `afterEach` isn't it?